src/ralph/backends/data/base.py

```diff
@@ -536,6 +548,31 @@ async def read(  # noqa: PLR0913
                 during encoding records and `ignore_errors` is set to `False`.
             BackendParameterException: If a backend argument value is not valid.
         """
+        if prefetch and prefetch != 1:
+            queue = Queue(prefetch - 1)
```

Comment:
why `prefetch - 1` ? If I understand correctly, this means we assume the function that consumes the generator starts consuming it immediately, which is not necessarily the case.

Reply:
Isn't the body of a generator not executed unless it is (started to be) consumed?